### PR TITLE
[FEATURE] Update commands to use `environment` over `ctx.obj`

### PR DESCRIPTION
**Pull request by mariahjrogers**
## Pull request checklist

Before submitting your PR, please review the following checklist:

- [ ] Consider adding a unit test if your PR resolves an issue.
- [x] All new and existing tests pass locally (`palm test`)
- [X] Lint (`palm lint`) has passed locally and any fixes were made for failures
- [X] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Breaking changes

- [x] Check if this pull request introduces a breaking change

## What does this implement/fix? Explain your changes.
Updated all instances of the `@pass_context` decorator to use `@pass_object` instead. Also updated any instances of `ctx.obj` to reference the context object `environment` for better code readability!

## Does this close any currently open issues?
#45 

## Any other comments?
…

## Where has this been tested?

**Operating System:** …

**Platform:** …

**Target Platform:** …

### CODE REVIEW

**Comment by jakeberesford-palmetto** [comment]
awwwww yeah! I was just thinking about how I wanted to do this soon!

File: palm/containerizer.py [hunk 1]
```diff
@@ -17,18 +17,18 @@ class Containerizer(ABC):
     Implementations of run() should call check_setup() and generate()
     """
 
-    def __init__(self, ctx, template_dir: Path) -> None:
+    def __init__(self, environment, template_dir: Path) -> None:
```
Comment (jakeberesford-palmetto) [inline palm/containerizer.py:20]: This is actually a breaking change, since Containerizer is part of the public API and we're changing it's arguments. I believe palm-dbt uses this class directly and would need an update to continue working...

**Comment by jakeberesford-palmetto** [comment]
I really wanted to be able to merge this right now, because it's a great change... but I need to do some thinking & looking around to see whether the change to Containerizer is a breaking change, I think it is... we might want to change our implementation there and see whether we can make it non-breaking, I'd rather not do a major version bump for this small of a changeset

File: palm/containerizer.py (hunk 1 above)
Comment (mariahjrogers) [reply palm/containerizer.py:20]: Gotcha, so what do you think should we do? Revert? Or make a cascading update?

Comment (jakeberesford-palmetto) [reply palm/containerizer.py:20]: https://github.com/palmetto/palm-dbt/blob/develop/palm/plugins/dbt/dbt_containerizer.py#L10

Subclassing PythonContainerizer... but not actually calling the parent constructor.

Still _technically_ a breaking change, even if the one external use I'm aware of may not actually break...

Comment (jakeberesford-palmetto) [reply palm/containerizer.py:20]: The problem with a cascading change is that we could break plugins we don't know about (I think Ethan had one or two...). We probably want to revert this change and:
1. add an optional kwarg for `environment`
2. logic to use `environment` over `ctx`
3. Add a deprecation warning if `ctx` has an `obj`
4. update cmd_containerize to something like `PythonContainerizer({}, template_dir, version,environment=environment).run()`

There may be a cleaner way... I'll think about it more tonight.

File: palm/environment.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ def run_in_shell(self, cmd: str, env_vars: Optional[dict] = {}):
         deprecation_msg = (
             "DEPRECATION: run_in_shell has been renamed to "
             "`run_in_docker` and will be removed in a future version. "
-            "Please update your commands to use ctx.obj.run_in_docker"
+            "Please update your commands to use environment.run_in_docker"
         )
         click.secho(deprecation_msg, fg="yellow")
```
Comment (jakeberesford-palmetto) [inline palm/environment.py:55]: If we go the deprecation route, we should make a `deprecation_message` utility that encapsulates some of this:
- receives a message arg
- prefixes with `DEPRECATION:`
- secho with `fg='yellow'`

Comment (mariahjrogers) [reply palm/environment.py:55]: I tried this, but as it turns out, this deprecation message is never being displayed... I couldn't get it to display using this deprecated `run_in_shell` function anywhere.

File: palm/plugins/core/commands/cmd_containerize.py [hunk 3]
```diff
@@ -12,9 +12,9 @@
...
 )
-@click.pass_context
-def cli(ctx, version: str):
+@click.pass_obj
+def cli(environment, version: str):
     all_templates_dir = Path(Path(__file__).parents[1], "templates")
     template_dir = all_templates_dir / "containerize"
-    PythonContainerizer(ctx, template_dir, version).run()
-    click.secho(f"Containerized {ctx.obj.palm.image_name}", fg="green")
+    PythonContainerizer(environment, template_dir, version).run()
```
Comment (jakeberesford-palmetto) [inline palm/plugins/core/commands/cmd_containerize.py:19]: This will fail with the reverted changes in containerizer.py

Let's revert the changes in this file too and make a GH issue tagged with v3 (next major version) to update this!

File: docs/source/code-generation.rst [hunk 4]
```diff
@@ -18,8 +18,8 @@ To use Palm code generation you will need:
    project's .palm directory.
 2. A YAML configuration file called ``template-config.yaml`` - this is a
    configuration file that describes how to generate code.
-3. A palm command which is decorated with the ``@click.pass_context`` decorator and
-   calls ``ctx.obj.generate(template_path, output_path, replacements)``.
+3. A palm command which is decorated with the ``@click.pass_obj`` decorator and
+   calls ``environment.generate(template_path, output_path, replacements)``.
```
Comment (jakeberesford-palmetto) [inline docs/source/code-generation.rst:22]: This change to docs also needs to be reverted

File: docs/source/high-level-features.rst [hunk 5]
```diff
@@ -28,7 +28,7 @@ Code generation
 
 Palm includes code generation functionality, powered by jinja. Code generation
 allows you to automate repetitive boiler plating tasks, and keep your codebase
-consistent. Code generation is driven by the ``ctx.obj.generate()`` function.
+consistent. Code generation is driven by the ``environment.generate()`` function.
```
Comment (jakeberesford-palmetto) [inline docs/source/high-level-features.rst:31]: and here

File: docs/source/introduction/viewpoint.rst [hunk 6]
```diff
@@ -103,15 +103,15 @@ what if the fix was automated? Enter palm.
 
    #.palm/cmd_up.py
 
-   def cmd_up(ctx):
+   def cmd_up(environment):
```
Comment (jakeberesford-palmetto) [inline docs/source/introduction/viewpoint.rst:106]:
```suggestion
   @click.pass_obj
   def cmd_up(environment):
```

**Review by jakeberesford-palmetto — CHANGES_REQUESTED** [review]
Changes needed - see previous comments

File: docs/source/code-generation.rst (hunk 4 above)
Comment (mariahjrogers) [reply docs/source/code-generation.rst:22]: Why is this necessary? The code generation functions on `ctx.obj` which is equivalent to `environment`. If we change this documentation then it won't reflect any of the commands that this PR updates to use `pass_obj` and `environment` over `ctx.obj`.

This `generate` function is not the Containerizer's `generate` function, it's the one for the `Environment` class. We reverted my change to Containerizer because it would have prevented other things that use `ctx` from working, but `Environment` isn't reverted.

Comment (jakeberesford-palmetto) [reply docs/source/code-generation.rst:22]: yup, you're right - fine to keep this as-is

File: docs/source/high-level-features.rst (hunk 5 above)
Comment (mariahjrogers) [reply docs/source/high-level-features.rst:31]: Same as above that was actually an Environment `generate` not a Containerize `generate`?

**Review by jakeberesford-palmetto — APPROVED** [review]
